hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -257,4 +261,40 @@ public ClientMessage getPayloadData(byte[] payload) {
         return this;
     }
 
+    @Override
+    public boolean writeTo(ByteBuffer destination) {
+        byte[] byteArray = buffer().byteArray();
+        int size = byteArray.length;
+        if (valueOffset >= size) {
```

Comment:
I took a second look at the code. It can be at most size. It can not be greater. And even equal check seems unnecessary. I will update